src/genie/libs/parser/iosxr/show_environment.py

```diff
@@ -0,0 +1,420 @@
+                          '\s+(?P<worst_case_power_used>[\d\.]+)$')
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            # R/S/I   Modules         Capacity        Status
+            # R/S/I           Power Supply    Voltage         Current
+            # R/S/I           Power Draw      Voltage         Current
+            m = p1.match(line)
+            if m:
+                # Parse section
+                section = m.groupdict()['section']
+                if section in ('Modules', 'Power Supply'):
+                    slot_type = "PSM"
+                elif section in ('Power Draw',):
+                    slot_type = "other_slot"
+                    #psm: power_supply_module, ft:fan_tray, rsp:rsp, oc: other_card, FC: fabric_card
+                continue
+
+            # 0/PS0/M0/*
+            # 0/PS0/M2/*
+            # 0/PS1/M0/*
+            # 0/PS1/M2/*
+            m = p2.match(line)
+            if m:
+                # Parse PSM rack_id, slot, full_slot
+                if slot_type == "PSM":
+                    group_dict = m.groupdict()
+                    psm_location = group_dict['psm_location']
+                    rack_id = group_dict['rack_id']
+                    slot_id = group_dict['slot_id']
+                    return_dict['rack'].setdefault(rack_id, {}).setdefault("slot", {}).setdefault(slot_type, {}).setdefault(slot_id, {}).setdefault("full_slot", psm_location)
+                    continue
+
+            # host    PM      4400            Ok
+            # host    PM      2200            Ok(Feeder A is missing)
+            m = p3.match(line)
+            if m:
+                # Parse psm capacity, status
+                if slot_type == "PSM":
+                    group_dict = m.groupdict()
+                    psm_capacity = group_dict['psm_capacity']
+                    psm_status = group_dict['psm_status']
+                    return_dict['rack'][rack_id]["slot"][slot_type][slot_id]["capacity"] = psm_capacity
+                    return_dict['rack'][rack_id]["slot"][slot_type][slot_id]["status"] = psm_status
+                    continue
+
+            # 0/PS0/M0/*       922.3          54.9            16.8
+            # 0/PS0/M2/*       880.0          55.0            16.0
+            # 0/PS1/M0/*       473.0          55.0             8.6
+            # 0/PS1/M2/*       434.5          55.0             7.9
+            # 0/PS0/*          922.3          54.9            16.8
+            # 0/PS2/*          880.0          55.0            16.0
+            # 0/PS4/*          473.0          55.0             8.6
+            # 0/PS6/*          434.5          55.0             7.9
+            # 0/PM0/*          922.3          54.9            16.8
+            # 0/PM2/*          880.0          55.0            16.0
+            # 0/PM4/*          473.0          55.0             8.6
+            # 0/PM6/*          434.5          55.0             7.9
```

Comment:
Thank you for adding plenty of comments, but lets remove some repetition 
```suggestion
            # 0/PS0/M0/*       922.3          54.9            16.8
            # 0/PS0/*          922.3          54.9            16.8
            # 0/PM0/*          922.3          54.9            16.8
```